### PR TITLE
Add an example for the embed_migrations macro.

### PR DESCRIPTION
**Pull request by Eijebong**

### CODE REVIEW

**Comment by dbrgn** [comment]
This doesn't seem to work:

```
error[E0308]: mismatched types
  --> src/db/mod.rs:69:77
   |
69 |         Ok(connection) => embedded_migrations::run_with_output(&connection, io::stdout())
   |                                                                             ^^^^^^^^^^^^ expected mutable reference, found struct `std::io::Stdout`
   |
   = note: expected type `&mut std::io::Write`
              found type `std::io::Stdout`
```

Instead, you need to pass a mut reference:

```
embedded_migrations::run_with_output(&connection, &mut std::io::stdout());
```

How come this wasn't caught by the test suite?

Also, maybe replace the `Example:` headline with `# Examples`? See https://github.com/diesel-rs/diesel/pull/840#pullrequestreview-31039462

**Comment by dbrgn** [comment]
Something else I noted: When using `run_with_output` instead of `run`, rust complains about the `run` function not being used:

```
warning: function is never used: `run`, #[warn(dead_code)] on by default
  --> src/db/mod.rs:62:1
   |
62 | embed_migrations!();
   | ^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate
```

**Comment by dbrgn** [comment]
Something else I noted: When using `run_with_output` instead of `run`, rust complains about the `run` function not being used:

```
warning: function is never used: `run`, #[warn(dead_code)] on by default
  --> src/db/mod.rs:62:1
   |
62 | embed_migrations!();
   | ^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate
```

Maybe @weiznich knows how to deal with the warning?

**Comment by Eijebong** [comment]
Yeah the unused function makes sense, maybe we should add an `#[allow(dead_code)]` on it.

File: diesel/src/macros/macros_from_codegen.rs [hunk 1]
```diff
@@ -72,6 +72,18 @@ macro_rules! infer_table_from_schema {
 ///
 /// This macro can only be used in combination with the `diesel_codegen` or
 /// `diesel_codegen_syntex` crates. It will not work on its own.
+///
+/// Example:
+/// ```
```
Comment (killercup) [inline diesel/src/macros/macros_from_codegen.rs]: Please add an empty line above code blocks, and below headlines. Bonus points for adding lang annotations to all code blocks, even though `rust` is the default (just my personal preference) :)

File: diesel/src/macros/macros_from_codegen.rs [hunk 2]
```diff
@@ -72,6 +72,18 @@ macro_rules! infer_table_from_schema {
 ///
 /// This macro can only be used in combination with the `diesel_codegen` or
 /// `diesel_codegen_syntex` crates. It will not work on its own.
+///
+/// Example:
```
Comment (killercup) [inline diesel/src/macros/macros_from_codegen.rs]: Common style is

    # Examples

Comment (Eijebong) [inline diesel/src/macros/macros_from_codegen.rs]: Oh yeah, this looks way better :)

**Comment by killercup** [comment]
🎉

**Pull request merged**